### ISSUE TITLE
Room booking reminders

### ISSUE THREAD

**Issue by phaelar** [GovTechSG/butler-bot#19]
I think it would be useful to send reminders to the user 5 min before the time slot starts and some actions to acknowledge or cancel the booking.
Also, maybe reminders 5 min before it ends to wrap up the meeting soon.

There have been times where I've booked a room in advance and forgot to cancel the booking when the meeting has been postponed.

**Comment by edisonchee** [comment]
This is a great idea. Do you think it makes sense to release the booking if no confirmation is done?

**Comment by phaelar** [comment]
yeah, makes sense.
will probably have to explicitly inform the user about this behaviour when they complete their booking.

**Comment by edisonchee** [comment]
yeah not sure how we can communicate this to our users. I can imagine situations where people miss the notification to confirm and end up having the room released.

**Comment by languantan** [comment]
I guess so far we dun see people complain about +phaelar "stealing" rooms, i.e. people "blocking" other people's use is still minimal. Having the room auto release kinda punishes those who do use the meeting room vs the small majority who don't.

That said, maybe we can have a feature where u see an empty room and u can message the user who booked it/ask for approval

**Comment by yuhong90** [comment]
Agree that auto releasing a booking might end up creating an issue when users forget to confirm, eg. when they are rushing to the meeting.

Maybe we can start with a reminder of the upcoming booking with the option to release the booking? 

On a seperate note, 5mins might be too short a lead time for the releasing of the room to be meaningful, seeing that we don't allow booking of a room once the start time of a 30 mins block has passed.
